src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const initialObject = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(initialObject, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const value of action.keysToRemove) {
+          delete initialObject[value];
+        }
+        break;
+
+      case 'clear':
+        for (const key in initialObject) {
+          delete initialObject[key];
+        }
+        break;
```

Comment:
3.[CODE STYLE]: switch/case should always have default case for error handling.

Reply:
Agree with @rtkachuk-moon, let`s add default case